src/main/java/uk/sky/cirrus/locking/Lock.java

```diff
@@ -0,0 +1,174 @@
+package uk.sky.cirrus.locking;
+
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.exceptions.DriverException;
+import com.datastax.driver.core.exceptions.WriteTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.sky.cirrus.locking.exception.CannotAcquireLockException;
+import uk.sky.cirrus.locking.exception.CannotReleaseLockException;
+
+import java.time.Duration;
+import java.util.UUID;
+
+/**
+ * Class to acquire a lock on cassandra table for a single application instance.  Each instance is given a unique identifier
+ * and attempts to gain a lock on the table.  If no lock is currently acquired, the instance is given the lock on the table, otherwise
+ * the instance must wait until the lock has been relinquished. If a lock cannot be acquired within the configured timeout
+ * interval, an exception is thrown.
+ */
+public class Lock {
```

Comment:
Can we use lock and unlock to match https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/locks/Lock.html?